pennylane/wires.py

```diff
@@ -493,6 +493,186 @@ def unique_wires(list_of_wires):
+        >>> wires1.union(wires2)
+        Wires([1, 2, 3, 4, 5])
+        """
+        return Wires(set(self.labels) | set(other.labels))
+
+    def __or__(self, other):
+        """Return the union of the current Wires object and another Wires object.
+
+        Args:
+            other (Wires): Another Wires object to perform the union with.
+
+        Returns:
+            Wires: A new Wires object representing the union of the two Wires objects.
+
+        Raises:
+            TypeError: If `other` is not an instance of Wires.
+
+        **Example**
+
+        >>> wires1 = Wires([1, 2, 3])
+        >>> wires2 = Wires([3, 4, 5])
+        >>> wires1 | wires2
+        Wires([1, 2, 3, 4, 5])
+        """
+        if not isinstance(other, Wires):
+            raise TypeError(f"Can only do the union of Wires with Wires. Got {type(other)}.")
+        return self.union(other)
+
+    def intersection(self, other):
+        """Return the intersection of the current Wires object and another Wires object.
+
+        Args:
+            other (Wires): Another Wires object to perform the intersection with.
+
+        Returns:
+            Wires: A new Wires object representing the intersection of the two Wires objects.
+
+        Raises:
+            TypeError: If `other` is not an instance of Wires.
+
+        **Example**
+
+        >>> wires1 = Wires([1, 2, 3])
+        >>> wires2 = Wires([2, 3, 4])
+        >>> wires1.intersection(wires2)
+        Wires([2, 3])
+        """
+        return Wires(set(self.labels) & set(other.labels))
+
+    def __and__(self, other):
+        """Return the intersection of the current Wires object and another Wires object.
+
+        Args:
+            other (Wires): Another Wires object to perform the intersection with.
+
+        Returns:
+            Wires: A new Wires object representing the intersection of the two Wires objects.
+
+        Raises:
+            TypeError: If `other` is not an instance of Wires.
+
+        **Example**
+
+        >>> wires1 = Wires([1, 2, 3])
+        >>> wires2 = Wires([2, 3, 4])
+        >>> wires1 & wires2
+        Wires([2, 3])
+        """
+        if not isinstance(other, Wires):
+            raise TypeError(f"Can only do the intersection of Wires with Wires. Got {type(other)}.")
+        return self.intersection(other)
+
+    def difference(self, other):
+        """Return the difference of the current Wires object and another Wires object.
+
+        Args:
+            other (Wires): Another Wires object to perform the difference with.
+
+        Returns:
+            Wires: A new Wires object representing the difference of the two Wires objects.
+
+        Raises:
+            TypeError: If `other` is not an instance of Wires.
+
+        **Example**
+
+        >>> wires1 = Wires([1, 2, 3])
+        >>> wires2 = Wires([2, 3, 4])
+        >>> wires1.difference(wires2)
+        Wires([1])
+        """
+        return Wires(set(self.labels) - set(other.labels))
```

Comment:
Will this raise an error here when `other` isn't `Wires`? 🤔

Reply:
Same applies for the other methods you added.

Reply:
Oh woops, I thought I was testing for that. No it doesn't raise an error as far as I can tell. Let me fix that.